### ISSUE TITLE
Parser.parse() fails on trailing whitespace

### ISSUE THREAD

**Issue by jkubrynski** [jOOQ/jOOQ#6707]
When script contains empty line at the end DDLDatabase generator throws `C42000_NO_SUBCLASS` exception

**Comment by lukaseder** [comment]
Thanks for your report. I can reproduce this. Quite inconvenient!

**Comment by lukaseder** [comment]
The issue can be reproduced with the `Parser` API directly:

```java
DSL.using(configuration).parser().parse("select 0; select 1;  ");
```

**Comment by lukaseder** [comment]
This will be merged to 3.10.2 (#6710)

**Comment by rossetti** [comment]
Is this error related to this issue:

Exception in thread "main" org.jooq.impl.ParserException: C42000_NO_SUBCLASS: ARY KEY GENERATED ALWAYS AS IDENTITY (START WITH 1[*], INCREMENT BY 1),
	SIM_NAME VARCHAR(510) NOT NULL,
	MODEL_NAME VARCHAR(510) NOT
	at org.jooq.impl.ParserImpl$ParserContext.unexpectedToken(ParserImpl.java:6068)
	at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:555)
	at org.jooq.impl.ParserImpl.parse(ParserImpl.java:353)
	at org.jooq.impl.ParserImpl.parse(ParserImpl.java:345)
	at org.jooq.util.ddl.DDLDatabase.create0(DDLDatabase.java:129)
	at org.jooq.util.AbstractDatabase.create(AbstractDatabase.java:237)
	at org.jooq.util.AbstractDatabase.create(AbstractDatabase.java:226)
	at org.jooq.util.AbstractDatabase.getDialect(AbstractDatabase.java:208)
	at org.jooq.util.AbstractGenerator.logDatabaseParameters(AbstractGenerator.java:131)
	at org.jooq.util.JavaGenerator.generate(JavaGenerator.java:271)
	at org.jooq.util.GenerationTool.run(GenerationTool.java:642)
	at dbutil.TestCodeGeneration.jooqCodeGeneration(TestCodeGeneration.java:76)
	at dbutil.TestCodeGeneration.main(TestCodeGeneration.java:21)
My SQL is thus:

CREATE TABLE SIMULATION_RUN (
	ID INTEGER NOT NULL PRIMARY KEY GENERATED ALWAYS AS IDENTITY (START WITH 1, INCREMENT BY 1),
	SIM_NAME VARCHAR(510) NOT NULL,
	MODEL_NAME VARCHAR(510) NOT NULL,
	EXP_NAME VARCHAR(510) NOT NULL,
	EXP_START_TIME_STAMP TIMESTAMP,
	EXP_END_TIME_STAMP TIMESTAMP,
	NUM_REPS INTEGER NOT NULL CHECK (NUM_REPS >=1),
	LAST_REP INTEGER, 
	LENGTH_OF_REP DOUBLE, 
	LENGTH_OF_WARM_UP DOUBLE,
	HAS_MORE_REPS BOOLEAN,
	REP_ALLOWED_EXEC_TIME BIGINT,
	REP_INIT_OPTION BOOLEAN,
	RESET_START_STREAM_OPTION BOOLEAN,
	ANTITHETIC_OPTION BOOLEAN,
	ADV_NEXT_SUB_STREAM_OPTION BOOLEAN,
	NUM_STREAM_ADVANCES INTEGER
);
